src/test/java/org/opensearch/ad/transport/RCFPollingTests.java

```diff
@@ -11,54 +11,7 @@
 
 package org.opensearch.ad.transport;
 
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.doAnswer;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
-
-import java.io.IOException;
-import java.net.InetAddress;
-import java.net.InetSocketAddress;
-import java.util.Collections;
-import java.util.Optional;
-
-import org.junit.AfterClass;
-import org.junit.Before;
-import org.junit.BeforeClass;
-import org.opensearch.Version;
-import org.opensearch.action.ActionListener;
-import org.opensearch.action.support.ActionFilters;
-import org.opensearch.action.support.PlainActionFuture;
-import org.opensearch.ad.AbstractADTest;
-import org.opensearch.ad.TestHelpers;
-import org.opensearch.ad.cluster.HashRing;
-import org.opensearch.ad.common.exception.AnomalyDetectionException;
-import org.opensearch.ad.common.exception.JsonPathNotFoundException;
-import org.opensearch.ad.constant.CommonName;
-import org.opensearch.ad.ml.ModelManager;
-import org.opensearch.ad.ml.SingleStreamModelIdMapper;
-import org.opensearch.cluster.node.DiscoveryNode;
-import org.opensearch.cluster.service.ClusterService;
-import org.opensearch.common.io.stream.StreamInput;
-import org.opensearch.common.settings.Settings;
-import org.opensearch.common.transport.TransportAddress;
-import org.opensearch.core.xcontent.ToXContent;
-import org.opensearch.tasks.Task;
-import org.opensearch.transport.ConnectTransportException;
-import org.opensearch.transport.Transport;
-import org.opensearch.transport.TransportException;
-import org.opensearch.transport.TransportInterceptor;
-import org.opensearch.transport.TransportRequest;
-import org.opensearch.transport.TransportRequestOptions;
-import org.opensearch.transport.TransportResponse;
-import org.opensearch.transport.TransportResponseHandler;
-import org.opensearch.transport.TransportService;
-
-import test.org.opensearch.ad.util.FakeNode;
-import test.org.opensearch.ad.util.JsonDeserializer;
-
-import com.google.gson.Gson;
-import com.google.gson.GsonBuilder;
+/*
```

Comment:
whats the reason for commenting out this entire test class, if there are any blockers we should note them in the comments here and tag the related issue

Reply:
It has implementation of Fake node where all the implementation regarding Plugins is done. We need to open an another issue and create a test suite for extensions in order to make this test work.

Reply:
FakeNode has 2 methods which relies on plugins.
`discoveryNode` -> we can use extension node here?
`connectNodes` -> we can connect to extension node here?
Can you try the above?

Reply:
Also, do we have a new method `connectToNodeAsExtension` now in TransportService:
https://github.com/opensearch-project/OpenSearch/pull/6866

Reply:
sure

Reply:
Hey @owaiskazi19 apparently taskResourceTrackingService objects needs clusterSettings. And we need to provide SDKClusterSettings. We need to make changes on Opensearch for that.

Reply:
Also we need to implement below method for SDKClusterService

```
        clusterService.addStateApplier(transportService.getTaskManager());

```

Reply:
In below method we need to make changes on Opensearch end to make it compatible with SDKClusterService

```
transportListTasksAction = new TransportListTasksAction(
            clusterService,
            transportService,
            actionFilters,
            taskResourceTrackingService
        );
```

Reply:
Try to pass `null` for those, if not then we can always cover them during the work of integ tests for AD Extension